packages/discord.js/src/structures/client/Bitfield.ts

```diff
@@ -0,0 +1,23 @@
+import { BitFieldResolvable } from '../../typedefs/BitFieldResolvable';
+
+export class BitField<K extends string, S extends number | bigint = number> {
+	public constructor(bits?: BitFieldResolvable<K, S>) {
+		this.bitfield = this.resolve(bits);
+		this.FLAGS = {};
+	}
+	private defaultBit: number = 0;
+	public bitfield: number | bigint;
+	public FLAGS: any; //better type for this later.
+	public resolve(bit?: BitFieldResolvable<any, number | bigint>): number | bigint {
+		const { defaultBit } = this;
+		if (typeof bit === 'undefined') return defaultBit;
```

Comment:
```suggestion
		if (bit === undefined) return defaultBit;
```

Reply:
Why thumbs down @NotSugden?

Reply:
These are not equivalent. Using `typeof` on an undeclared variable will return the string `"undefined"`, making it more robust, while your suggested change would throw a TypeError.

Reply:
No it wouldn't.

Reply:
Why the 😕, I'm right:
```js
const {BitField} = require('./dist/structures/client/Bitfield');
b.resolve(); // returns 0, doesn't throw because bit is declared as a parameter
```

Reply:
Just because it isn't undeclared doesn't make my statement incorrect, or make this a useful change.
`typeof` is still a more robust check.

Reply:
Robust against what? The parameter is always there, comparing `bit` to `undefined` directly will *always* work. If the parameter is ever removed the compiler will prevent any bugs from actually being released, there is no reason to keep the `typeof` check.